unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -7275,6 +7223,133 @@ ProceduralAsset asset
             actionFinishedEmit(true, geoList);
         }
 
+        private Bounds GetAgentBounds(GameObject gameObject, Type agentType) {
+            Debug.Log(agentType);
+            Debug.Log(typeof(StretchAgentController));
+            Bounds bounds = new Bounds(gameObject.transform.position, Vector3.zero);
+            MeshRenderer[] meshRenderers = gameObject.GetComponentsInChildren<MeshRenderer>();
+            if (agentType == typeof(LocobotFPSAgentController)) {
+                meshRenderers = this.baseAgentComponent.BotVisCap.GetComponentsInChildren<MeshRenderer>();
+            } else if (agentType == typeof(StretchAgentController)) {
+                meshRenderers = this.baseAgentComponent.StretchVisCap.GetComponentsInChildren<MeshRenderer>();
+            }
+            foreach (MeshRenderer meshRenderer in meshRenderers) {
+                bounds.Encapsulate(meshRenderer.bounds);
+            }
+            return bounds;
+        }
+
+        public void spawnAgentBoxCollider(GameObject agent, Type agentType, Vector3 scaleRatio, bool useAbsoluteSize = false, bool useVisibleColliderBase = false) {
+            // Store the current rotation
+            Vector3 originalPosition = this.transform.position;
+            Quaternion originalRotation = this.transform.rotation;
+
+            // Move the agent to a safe place and align the agent's rotation with the world coordinate system
+            this.transform.position = new Vector3(originalPosition.x + 100, originalPosition.y + 100, originalPosition.z + 100);
```

Comment:
Would store the `100` in a constant variable for the class, and use it here and below, maybe a Vector3 `agentSpawnOffset` ?